draft-smyslov-ipsecme-ikev2-reliable-transport.xml

```diff
@@ -1,148 +1,220 @@
+        4500 directly, as specified in <xref target="RFC9329"></xref>, for
+        example, when large key exchange payloads (with large public keys) are
+        expected. In this case, the initiator includes the SEPARATE_TRANSPORTS
+        notification in the IKE_SA_INIT request to indicate its preference to
+        use separate transports; IKEv2 over TCP and ESP over UDP, provided
+        that UDP is not blocked in the network path. </t>
+
+        <t>If the responder supports this extension, it responds with the
+        SEPARATE_TRANSPORTS notification in the IKE_SA_INIT response. In this
+        case, child SAs are created as per <xref target="RFC7296"></xref>,
+        with ESP sent over UDP if possible, or over TCP as described in <xref
+        target="RFC9329"></xref> if UDP is blocked. </t>
+
+        <t>If the responder does not support the SEPARATE_TRANSPORTS
+        notification (i.e., the notification is not echoed in the response),
+        the initiator MUST treat this as an indication that the responder does
+        not support separate transports. In this case, both IKEv2 and ESP MUST
+        use TCP transport for all subsequent exchanges, as per <xref
+        target="RFC9329"></xref>.</t>
+
+        <t>In both scenarios, once the IKEv2 exchange switches to TCP
+        transport, either after IKE_SA_INIT or if TCP was used from the
+        beginning, all subsequent IKEv2 exchanges MUST continue to use
+        TCP.</t>
+      </section>
+
+      <section title="Notification Format">
+        <t>The SEPARATE_TRANSPORTS notification has Protocol ID set to 0 and
+        SPI Size set to 0. This specification does not define any notification
+        data, the notification is sent with no data. Future specifications may
+        define data for this notification. Peers conforming to this
+        specification MUST ignore any data if present.</t>
+      </section>
+
+      <section title="ESP Behavior">
+        <t>Child SAs are created as specified in <xref
+        target="RFC7296"></xref>. ESP packets either use direct transport over
+        IP or are UDP encapsulated if NAT is detected. If UDP transport for
+        ESP becomes unavailable (e.g., blocked by a firewall), peers MAY
+        switch ESP to use TCP transport as specified in <xref
+        target="RFC9329"></xref>. Peers are responsible for maintaining NAT
+        mappings by sending NAT keepalive packets (see Section 2.23 of <xref
+        target="RFC7296"></xref>) when ESP over UDP is used. If ESP is
+        transported over a different protocol than IKE, intermediate devices
+        might apply different filtering rules. To detect possible connectivity
+        issues with ESP traffic, the encrypted ESP ping mechanism defined in
+        <xref target="I-D.ietf-ipsecme-encrypted-esp-ping"></xref> MAY be
+        used.</t>
+
+        <t><figure>
+            <artwork align="center"><![CDATA[
```

Comment:
This figure is not for Child SA creation, it illustrates negotiation of the feature in the IKE_SA_INIT. I think that it should be moved closer to the text that describes this process. Since you now separated cases when IKE_SA_INIT is started over UDP and over TCP into different sections, then either this picture should be duplicated (I don't like this) or put into its own section right after section about TCP in IKE_SA_INIT.

Reply:
Added two separate figures for Section 3.1 and Section 3.2 to keep it simple.